### ISSUE TITLE
Add logout-uri attribute

### ISSUE THREAD

**Issue by andreareginato** [angularjs-oauth/oauth-ng#12]
We need to add the logout-uri attribute to associate a URI that will be called when the widget logs out. Its logic should be pretty similar to the one used for the profile-uri.

**Comment by sguilly** [comment]
+1

**Comment by tomas487** [comment]
+1

**Comment by GregTurner** [comment]
+1

**Comment by andreareginato** [comment]
Anyone willing on working on this? It would be really appreciated.

**Comment by GregTurner** [comment]
Unfortunately OAuth2 doesn't specify logout. OpenID Connect does. Check out
RP initiated logout:

http://openid.net/specs/openid-connect-session-1_0.html

**Comment by andreareginato** [comment]
I think we can put into our "implementation". It's a must to have feature,
at least from my point of view.

On Tue, Jul 7, 2015 at 4:35 AM, Greg Turner <notifications@github.com>
wrote:

> Unfortunately OAuth2 doesn't specify logout. OpenID Connect does. Check out
> RP initiated logout:
>
> http://openid.net/specs/openid-connect-session-1_0.html
> On 6 Jul 2015 5:47 am, "Andrea Reginato" <notifications@github.com> wrote:
>
> > Anyone willing on working on this? It would be really appreciated.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/andreareginato/oauth-ng/issues/12#issuecomment-118843551
> >
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/andreareginato/oauth-ng/issues/12#issuecomment-119052136>
> .
>



-- 
Andrea Reginato
Lelylan | reThink your house
http://lelylan.com

**Comment by GregTurner** [comment]
I don't disagree it should be implemented.  Maybe just denote it as an
extension to a standard "x-logoutUri".

**Comment by andreareginato** [comment]
I agree with your point @GregTurner. I think the `x-` prefix could be a nice to have, to make it clear it's an addition "we" propose.

**Comment by lehne** [comment]
+1

**Comment by bramski** [comment]
Did this go anywhere?

**Comment by andreareginato** [comment]
Not yet. No pull request on it at the moment.

**Comment by phillip-hall** [comment]
I have just forked and will work on this as I need it in my project.

**Comment by bramski** [comment]
I have this in the wings and pretty much done.  Can make a PR soon-ish.

**Comment by bramski** [comment]
https://github.com/bramski/oauth-ng has most of what you need right now @phillip-hall.  Needs additional tests added to make it PR-able.

**Comment by phillip-hall** [comment]
OK, I will take a look at this.  thanks

**Comment by bramski** [comment]
This has now been merged to 0.4.7.
